cis194/week4/zehua/test/ExercisesSpec.hs

```diff
@@ -29,3 +48,59 @@ spec = do
       mapM_ (liftM2 shouldBe fun2' fun2)
         [2..10]
 
+  describe "ex2" $ do
+    it "works with empty" $ do
+      foldTree ([] :: [Char]) `shouldBe` (Leaf :: Tree Char)
+
+    it "works with one item" $ do
+      foldTree "A" `shouldBe` (Node 0 Leaf 'A' Leaf)
+
+    it "works with multiple items" $ do
+      mapM_ (\l -> foldTree l `shouldSatisfy` isBalancedTree)
+        [ "AB"
+        , "ABCDEFG"
+        , "ABCDEFGHIJKLMNOPQRSTUVWXYZ" ]
+
+  describe "ex3" $ do
+    it "xor" $ do
+      xor [] `shouldBe` False
+      xor [True] `shouldBe` True
+      xor [True, False, True] `shouldBe` False
+      xor [True, False, True, False] `shouldBe` False
+      xor [True, False, True, False, False] `shouldBe` False
+      xor [True, False, True, True, False, False] `shouldBe` True
```

Comment:
anyway to lift this into something to accept
```
[ (False, [])
, (True, [True])
, ... ]
```

Reply:
try to use `mapM_` and `uncurry` on the `shouldBe`

Reply:
`uncurry` with `shouldBe` seems not that intuitive to understand. resorting to lambda expression.